### PR TITLE
317: Ensure data submitted on frontend makes it to the backend

### PR DESCRIPTION
**Pull request by jasnakai**
This PR ensures that the user can perform the pre-SAC steps of checking eligibility, checking UEI, and adding audit access information. ([GSA-TTS/FAC#317](https://github.com/GSA-TTS/FAC/issues/317))

Submission criteria check:
- Changed the radio button input values to booleans from strings

Auditee information
- Updated UEI search input `name` to match `id`

Audit submission access:
- Removed Name inputs
- Give additional contacts a unique `id` and `name`
- Prepare form data to match API expectation

## Walkthrough of functionality

The user must first log into login.gov.
Screen Shot 2022-07-26 at 9 24 37 AM

Successful login will take the user to the first step, Submission criteria check.
Screen Shot 2022-07-26 at 9 28 50 AM

On Continue, the user is taken to the second step, Auditee information. The user can check their UEI here and add in the fiscal start and end dates for the submission.
Screen Shot 2022-07-26 at 9 26 29 AM

On successful submission, the user is taken to the third step, Audit submission access. Here, the user adds the contact information for the Auditee, Auditor and any other contacts needed.
Screen Shot 2022-07-26 at 9 28 25 AM

On successful submission, the user is taken to the audit submission page
Screen Shot 2022-07-26 at 9 52 41 AM
.

### CODE REVIEW

**Comment by jasnakai** [comment]
Updating tests.

**Comment by jasnakai** [comment]
[Preview](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/jn-317-access-form-febe-integration/audit/new/step-1/)

**Comment by jasnakai** [comment]
@petermarks-gsa Hey Peter, the PR is ready for review.

File: src/js/check-access.js [hunk 1]
```diff
@@ -1,6 +1,66 @@
...
+  );
+  preparedData['auditor_contacts'] = contactsToArray(
+    formData,
+    'auditor_contacts_email'
+  );
+
+  const headers = new Headers();
+  headers.append('Content-type', 'application/json');
+
+  getApiToken().then((token) => {
```
Comment (heymatthenry) [inline src/js/check-access.js:28]: Not a blocker, but there's a [`queryApi`](https://github.com/GSA-TTS/FAC-Frontend/blob/2dbf04b1776bb002ad2a8f054a85b5a05c173665/src/js/api.js#L3) method that does this, so there's no need to keep writing this for every API call.

**Review by heymatthenry — APPROVED** [review]
@jasnakai One non-blocking comment, but otherwise LGTM.

@petermarks-gsa I added this branch's Federalist preview URL to the Login dashboard so you should be able to do QA there.

File: src/js/check-access.js [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+    })
+      .then((resp) => resp.json())
+      .then((data) => {
+        handleAuditeeResponse(data);
+      });
+  });
+}
+
+function handleAuditeeResponse(data) {
+  const nextUrl = '/audit/submission/'; //data.next; //URL value for now
```
Comment (geekygirlsarah) [inline src/js/check-access.js:44]: [NB] Not sure if this comment should be removed or not?

**Review by geekygirlsarah — APPROVED** [review]
👍 from me but might want a frontend dev to also review

File: src/js/check-access.js (hunk 1 above)
Comment (jasnakai) [reply src/js/check-access.js:28]: Doh! Let me make that update.

**Review by petermarks-gsa — CHANGES_REQUESTED** [review]
@jasnakai  @heymatthenry @geekygirlsarah Here are my end-to-end QA results: 

* ✅ Start at [FAC Homepage](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/jn-317-access-form-febe-integration/)
* ✅ Login to existing Login.gov account with peter.marks@gsa.gov
* ❓Redirected to blank "My audit submissions" page instead of step 1 - is this because I have an audit in progress?
* ✅ Manually go to [Step 1](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/jn-317-access-form-febe-integration/audit/new/step-1/), fill out form. 
* ✅ Fill out step 2 with valid UEI (ZQGGHJH74DW7), progress to next form
* ✅ Fill out step 3 with incomplete data, see inline errors as expected
* ❌ Stay on the page, correct errors, press submit - Cannot progress to the next page. Invalid state not letting me? I was able to get around this by going back through steps 1-2 and filling out completed data on first pass of step 3. 
* ❌ On successful submission, the user is taken to the audit submission page - I was taken to 404 after successful submission - related to me having an audit in progress? 
* ❓How can I verify data was saved in django admin?

**Comment by jasnakai** [comment]
@petermarks-gsa The redirect after login might be a preview issue. It redirects correctly for me locally.
I will look into why the previous form data is being lost on a failed submission on form 3 (Audit submission access).
I do not know about the django admin piece.

**Comment by petermarks-gsa** [comment]
@jasnakai Thanks, I'm not sure I understand exactly what you mean by preview issue. My expectation is that once we've done the manual steps on federalist @tadhg-ohiggins is documenting in https://github.com/GSA-TTS/FAC/issues/314, we should be able to demo everything in Milestone 1 there end to end (login, three forms, verify data in django admin). I'm not sure if that is out of scope for this exact PR, but that's what I had in mind for Milestone 1 acceptance criteria.

**Comment by jasnakai** [comment]
@petermarks-gsa I have addressed the issues you mentioned in your previous review. Please let me know if you run into any of those or other issues.

**Review by petermarks-gsa — APPROVED** [review]
@jasnakai All green now! 

* ✅ Start at [FAC Homepage](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/preview/gsa-tts/fac-frontend/jn-317-access-form-febe-integration/)
* ✅ Login to existing Login.gov account with peter.marks@gsa.gov
* ✅ New redirected to step 1 as expected, fill out form. 
* ✅ Fill out step 2 with valid UEI (ZQGGHJH74DW7), progress to next form
* ✅ Fill out step 3 with incomplete data, see inline errors as expected
* ✅ Stay on the page, correct errors, press submit, now on general information page as expected
* ✅ Verify data was saved in django admin

**Pull request merged**